cdap-app-fabric/src/main/java/co/cask/cdap/test/internal/InMemoryNamespaceClient.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.test.internal;
+
+import co.cask.cdap.common.exception.CannotBeDeletedException;
+import co.cask.cdap.common.exception.NotFoundException;
+import co.cask.cdap.common.exception.UnauthorizedException;
+import co.cask.cdap.common.namespace.AbstractNamespaceClient;
+import co.cask.cdap.internal.app.namespace.NamespaceAdmin;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.NamespaceMeta;
+import co.cask.common.http.HttpRequest;
+import co.cask.common.http.HttpResponse;
+import com.google.inject.Inject;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+
+/**
+ *
+ */
+public class InMemoryNamespaceClient extends AbstractNamespaceClient {
```

Comment:
This isn't really in-memory, no. You're bypassing the remote call through other namespace clients, so maybe LocalNamespaceClient. Also, empty javadoc.